File: TransCoder_tokenized_test_set_functions/python/RETURN_PREVIOUS_ELEMENT_IN_AN_EXPANDING_MATRIX.py
def findLeft ( str ) :
    n = len ( str ) - 1;
    
    while ( n > 0 ) :
        if ( str [ n ] == 'd' ) :
            str = str [ 0 : n ] + 'c' + str [ n + 1 : ];
            
            break;
            
        if ( str [ n ] == 'b' ) :
            str = str [ 0 : n ] + 'a' + str [ n + 1 : ];
            
            break;
            
        if ( str [ n ] == 'a' ) :
            str = str [ 0 : n ] + 'b' + str [ n + 1 : ];
            
        elif ( str [ n ] == 'c' ) :
            str = str [ 0 : n ] + 'd' + str [ n + 1 : ];
            
        n -= 1;
        
    return str;

File: TransCoder_tokenized_test_set_functions/java/RETURN_PREVIOUS_ELEMENT_IN_AN_EXPANDING_MATRIX.java
static StringBuilder findLeft ( StringBuilder str ) {
  int n = str . length ( );
  while ( n > 0 ) {
    n --;
    if ( str . charAt ( n ) == 'd' ) {
      str . setCharAt ( n, 'c' );
      break;
    }
    if ( str . charAt ( n ) == 'b' ) {
      str . setCharAt ( n, 'a' );
      break;
    }
    if ( str . charAt ( n ) == 'a' ) str . setCharAt ( n, 'b' );
    else if ( str . charAt ( n ) == 'c' ) str . setCharAt ( n, 'd' );
  }
  return str;
}